smack-extensions/src/main/java/org/jivesoftware/smackx/avatar/UserAvatarManager.java

```diff
@@ -0,0 +1,312 @@
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.WeakHashMap;
+
+import org.jivesoftware.smack.ConnectionCreationListener;
+import org.jivesoftware.smack.Manager;
+import org.jivesoftware.smack.XMPPConnection;
+import org.jivesoftware.smack.XMPPConnectionRegistry;
+import org.jivesoftware.smack.SmackException.NoResponseException;
+import org.jivesoftware.smack.SmackException.NotConnectedException;
+import org.jivesoftware.smack.XMPPException.XMPPErrorException;
+import org.jivesoftware.smack.util.SHA1;
+import org.jivesoftware.smack.util.stringencoder.Base64;
+import org.jivesoftware.smackx.avatar.element.DataExtension;
+import org.jivesoftware.smackx.avatar.element.MetadataExtension;
+import org.jivesoftware.smackx.disco.ServiceDiscoveryManager;
+import org.jivesoftware.smackx.disco.packet.DiscoverItems.Item;
+import org.jivesoftware.smackx.pubsub.LeafNode;
+import org.jivesoftware.smackx.pubsub.PayloadItem;
+import org.jivesoftware.smackx.pubsub.PubSubManager;
+
+/**
+ * User Avatar manager class.
+ * 
+ * @author Fernando Ramirez
+ * @see <a href="http://xmpp.org/extensions/xep-0084.html">XEP-0084: User
+ *      Avatar</a>
+ */
+public final class UserAvatarManager extends Manager {
+
+    public static final String DATA_NAMESPACE = "urn:xmpp:avatar:data";
+    public static final String METADATA_NAMESPACE = "urn:xmpp:avatar:metadata";
+
+    static {
+        XMPPConnectionRegistry.addConnectionCreationListener(new ConnectionCreationListener() {
+            @Override
+            public void connectionCreated(XMPPConnection connection) {
+                getInstanceFor(connection);
+            }
+        });
+    }
+
+    private static final Map<XMPPConnection, UserAvatarManager> INSTANCES = new WeakHashMap<>();
+
+    /**
+     * Get the singleton instance of UserAvatarManager.
+     *
+     * @param connection
+     * @return the instance of UserAvatarManager
+     */
+    public static synchronized UserAvatarManager getInstanceFor(XMPPConnection connection) {
+        UserAvatarManager userAvatarManager = INSTANCES.get(connection);
+
+        if (userAvatarManager == null) {
+            userAvatarManager = new UserAvatarManager(connection);
+            INSTANCES.put(connection, userAvatarManager);
+        }
+
+        return userAvatarManager;
+    }
+
+    private UserAvatarManager(XMPPConnection connection) {
+        super(connection);
+    }
+
+    /**
+     * Returns true if User Avatar is supported by the server.
+     * 
+     * @return true if User Avatar is supported by the server.
+     * @throws NoResponseException
+     * @throws XMPPErrorException
+     * @throws NotConnectedException
+     * @throws InterruptedException
+     */
+    public boolean isSupportedByServer()
+            throws NoResponseException, XMPPErrorException, NotConnectedException, InterruptedException {
+        return findNodeItem(DATA_NAMESPACE) && findNodeItem(METADATA_NAMESPACE);
```

Comment:
I don't think that this is how you determine if xep84 is supported by the server. You likely just need to check if PEP is supported.